lib/custom_wizard/action.rb

```diff
@@ -133,6 +133,31 @@ def update_profile
     end
   end
 
+  def watch_categories
+
+    watched_categories = CustomWizard::Mapper.new(
+      inputs: action['categories'],
+      data: data,
+      user: user
+    ).perform
+
+    notification_level = action['notification_level']
+
+    mute_remainder = CustomWizard::Mapper.new(
+      inputs: action['mute_remainder'],
+      data: data,
+      user: user
+    ).perform
+
+    Category.all.each do |category|
+      if watched_categories.include?(category.id.to_s)
```

Comment:
You need to check for the presence of any watched categories (at all) before this condition, otherwise an exception is thrown: ``NoMethodError (undefined method `include?' for nil:NilClass)``.

Reply:
Resolved and added one for notification_level